### PR TITLE
Require a specific commit of lightning

### PR DESCRIPTION
**Pull request by erick-xanadu**
**Context:** This is a nice change as it allows a user to avoids developers pennylane-lightning manually.

**Description of the Change:**

**Benefits:** `pip install -e .` installs specific versions of pennylane and pennylane-lightning.

**Possible Drawbacks:** It would be better if the runtime's Makefile also looked into the `dep-version` file.

**Related GitHub Issues:**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -38,9 +38,13 @@
     pl_str = "pennylane="
     pl_str_length = len(pl_str)
     pl_version = [line[pl_str_length:].strip() for line in lines if pl_str in line][0]
+    lightning_str = "lightning="
+    lightning_str_length = len(pl_str)
+    lightning_version = [line[lightning_str_length:].strip() for line in lines if lightning_str in line][0]
 
 requirements = [
     f"pennylane @ git+https://github.com/pennylaneai/pennylane@{pl_version}",
+    f"pennylane-lightning @ git+https://github.com/pennylaneai/pennylane-lightning@{lightning_version}",
```
Comment (grwlf) [inline setup.py:47]: Are we sure that `pip` install pennylane-lightning _after_ the `pennylane`? Can PL overwrite PL-lightning if the order happens to be inversed?

Comment (dime10) [reply setup.py:47]: Hmm good question, I don't know if pip enforces an order here. I think it collects all dependencies recursively first, then attempts to match the version restrictions, before installing each package individually. So I don't think the order in this list should matter.

File: .github/workflows/check-catalyst.yaml [hunk 2]
```diff
@@ -378,7 +378,7 @@ jobs:
     - name: Install Deps
       run: |
         sudo apt-get update
-        sudo apt-get install -y python3 python3-pip libomp-dev libasan6
+        sudo apt-get install -y python3 python3-pip libomp-dev libasan6 ninja-build make cmake clang libomp-dev
```
Comment (maliasadi) [inline .github/workflows/check-catalyst.yaml:381]:
```suggestion
        sudo apt-get install -y python3 python3-pip libomp-dev libasan6 ninja-build make cmake clang
```

File: .github/workflows/check-catalyst.yaml [hunk 3]
```diff
@@ -450,7 +450,7 @@ jobs:
...
```
Comment (maliasadi) [inline .github/workflows/check-catalyst.yaml:453]:
```suggestion
        sudo apt-get install -y python3 python3-pip libomp-dev libasan6 ninja-build make cmake clang
```

File: .github/workflows/check-catalyst.yaml [hunk 4]
```diff
@@ -504,7 +504,7 @@ jobs:
...
```
Comment (maliasadi) [inline .github/workflows/check-catalyst.yaml:507]:
```suggestion
        sudo apt-get install -y python3 python3-pip libomp-dev libasan6 ninja-build make cmake clang
```

File: .dep-versions [hunk 5]
```diff
@@ -3,4 +3,5 @@ jax=0.4.23
 mhlo=4611968a5f6818e6bdfb82217b9e836e0400bba9
 llvm=cd9a641613eddf25d4b25eaa96b2c393d401d42c
 enzyme=1beb98b51442d50652eaa3ffb9574f4720d611f1
-pennylane=95129a0d6365b48cb4acfa828ceb6a8532e47ef5
+pennylane=cb6f1cdb8cda09eae9113284eb12b9dde18544a2
+lightning=efb946a736f824835ebf4fd692568d49e6cf1ca5
```
Comment (maliasadi) [inline .dep-versions:7]: Would you update the runtime to also use this commit from `.dep-versions` to build LQ & LK backend devices?

**Review by maliasadi — COMMENTED** [review]
:partying_face:

**Comment by dime10** [comment]
closed in favour of #568

File: setup.py (hunk 1 above)
Comment (grwlf) [reply setup.py:47]: Yes, I agree. Indeed, the race is not happening inside pip. It is between pip and our `make runtime` command. `make runtime` happens before `make frontend`, so I think we need to check this place actually.

**Pull request closed without merging**